### DISCUSSION TITLE
How to properly format markdown with more than 3 header levels?

### DISCUSSION THREAD

**Discussion by nonathaj** [withastro/starlight#1904] [Q&A] [answered]
I maintain a documentation website that has several different sources of generated markdown files.  Those generated files use many more header levels (some as deep as 6) than the default starlight configuration uses ([seems to be 3 from the documentation](https://starlight.astro.build/guides/authoring-content/#automatic-heading-anchor-links)).  

Is there a way for starlight to properly render headers beyond level 3?  I don't need it to be generated as part of the sidebar (though that would be cool to have the option).

```
# Level 1 header - works fine and appears on sidebar

## Level 2 header - works fine and appears on sidebar

### Level 3 header - works fine and appears on sidebar

#### Level 4 header - line is rendered as plain text and does NOT appear on sidebar

##### Level 5 header - line is rendered as plain text and does NOT appear on sidebar
```

**Comment by HiDeoo** [comment, marked as answer]
👋 Two routes would be available to address your issue:

- Configuring this globally using the [`tableOfContents`](https://starlight.astro.build/reference/configuration/#tableofcontents) option which let you configure the minimum and maximum level to include in the table of content for all pages.
- Configuring this per page using the frontmatter [`tableOfContents`](https://starlight.astro.build/reference/frontmatter/#tableofcontents) option.

It sounds like you may be more interested by the first solution if you have many pages so [editing your `astro.config.mjs`](https://starlight.astro.build/reference/configuration/#configure-the-starlight-integration) to update your Starlight configuration to the following will do the trick:

```js
export default defineConfig({
  integrations: [
    starlight({
      title: 'My docs',
      // Update the table of contents minimum and maximum heading level.
      tableOfContents: { minHeadingLevel: 2, maxHeadingLevel: 6 }
    }),
  ],
});
```

**Reply by nonathaj** [reply]
Hey HiDeoo, appreciate the speedy response.

That worked for getting them to appear in the sidebar.  While trying that out, we figured out the issue with rendering was on our side.  We had a `<br/>` on the line before a bunch of those headers that was causing the headings to render as normal paragraphs.

So I should be all set here.  Thanks again!
